content/en/blog/_posts/2025-04-23-jobs-backofflimitperindex-goes-ga.md

```diff
@@ -0,0 +1,103 @@
+---
+layout: blog
+title: "Kubernetes 1.33: Job's Backoff Limit Per Index Goes GA"
+date: 2025-04-23
+slug: kubernetes-1-33-jobs-backoff-limit-per-index-goes-ga
+author: >
+  [Michał Woźniak](https://github.com/mimowo) (Google)
+---
+
+In Kubernetes v1.33, the _Backoff Limit Per Index_ feature reaches general
+availability (GA). This blog describes the Backoff Limit Per Index feature and
+its benefits.
+
+## About Backoff Limit Per Index
+
+When you run workloads on Kubernetes, you must consider scenarios where Pod
+failures can affect the completion of your workloads. Ideally, your workload
+should tolerate transient failures and continue running.
+
+To achieve failure tolerance in Kubernetes Job, you can set the
+`spec.backoffLimit` field. This field specifies the total number of tolerated
+failures.
+
+However, for workloads where every index is considered independent, like
+[embarassingly parallel](https://en.wikipedia.org/wiki/Embarrassingly_parallel)
+workloads, the `spec.backoffLimit` field is often not flexible enough if you use
+Indexed Jobs. For example, integration tests. In that scenario, a fast-failing
+index is likely to consume your entire budget for tolerating Pod failures, and
+you might not be able to run the other indexes.
+
+In order to address this limitation, we introduce _Backoff Limit Per Index_,
+which allows you to control the number of retries per index.
+
+## How Backoff Limit Per Index works
+
+To use Backoff Limit Per Index for Indexed Jobs, configure the following:
+* Specify the number of tolerated Pod failures per index with the
+  `spec.backoffLimitPerIndex` field. When you set this field, the Job executes
+  all indexes by default.
+* Specify the cap on the total number of failed indexes by setting the
+  `spec.maxFailedIndexes` field. When the limit is exceeded the entire Job is
+  terminated.
+* Short-circuit to detect a failed index by using the `FailIndex` action in the
+  [Pod Failure Policy](/docs/concepts/workloads/controllers/job/#pod-failure-policy)
+  feature.
```

Comment:
```suggestion
* Configure a short circuit to detect a failed index by using the `FailIndex` action in the
  [Pod Failure Policy](/docs/concepts/workloads/controllers/job/#pod-failure-policy)
  feature.
```

Reply:
This is just a nit if it makes sense to you.

Reply:
I just wanted to make the list of bullets parallel by using a verb at the begining: specify, specify,...
Maybe 'Define a short circuit to detect a failed index by using the `FailIndex` action in the'?

Again, this is such a nit, you can disregard this thread altogether 😄

Reply:
I see the point as we start the previous bullet point with Specify. Makes sense. I like Define. I will update it shortly. Thank you!

Reply:
Done